terraform/install.sh

```diff
@@ -58,6 +60,51 @@ getBillingAccount() {
   log "using billing account: $billing_acct"
 }
 
+getProject() {
+  log "Checking for project list"
+  found_projects=$(gcloud projects list --filter="name:stackdriver-sandbox-*" --format="value(projectId)")
+  if [ -z "$found_projects" ] || [[ ${#found_projects[@]} -eq 0 ]]; then
+    create_flag=true
+  else
+      log "Which project would you like to use?:"
+      IFS_bak=$IFS
+      IFS=$'\n'
+      select opt in ${found_projects} "create" "cancel"; do
+        if [[ "${opt}" == "cancel" ]]; then
+          exit 0
+        elif [[ "${opt}" == "create" ]]; then
+          create_flag=true
+          log "create a new Sandbox!"
+          break
+        elif [[ -z "${opt}" ]]; then
+          log "invalid response"
+        else
+          project_id=$opt
+          bucket_name="$project_id-bucket"
+          break
+        fi
+      done
+      IFS=$IFS_bak
+  fi
+}
+
+createProject() {
+  if [ "$create_flag" = true ]; then
+    # generate random id
```

Comment:
The degree of randomness should not exceed maximum allowed name length for project name. Did you check the restrictions on the project id?

Reply:
The random number is 4 bytes, the same as previous.